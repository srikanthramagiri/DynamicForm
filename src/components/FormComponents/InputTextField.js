import React, { useState } from "react";
import { Form, Row, Col } from "react-bootstrap";
import Label from "./Label";
import AlertError from "./AlertError";
import PropTypes from "prop-types";

export default function InputTextField(props) {
  let { name, type, placeholder, label, changeHandler, validation } = props;
  let { regex, errorMessage } = validation ? validation : {};
  let [isvalid, setIsValid] = useState(true);
  const inputChangeHandler = (e) => {
    setIsValid(true);
  };

  const inputBlurHandler = (e) => {
    if (e.target.value === "") {
      setIsValid(false);
    }
    let reg = new RegExp(regex);
    setIsValid(reg.test(e.target.value));
    changeHandler(e);
  };

  return (
    <Form.Group controlId={name}>
      <Row>
        <Col lg={6} md={6}>
          <Label name={name} label={label} />
        </Col>
        <Col lg={6} md={6}>
          <Form.Control
            type={type}
            data-test='input-text-field'
            name={name}
            placeholder={placeholder}
            onChange={(e) => inputChangeHandler(e)}
            onBlur={(e) => inputBlurHandler(e)}
            max={100}
          />
        </Col>
        <Col lg={6} />
        <Col lg={6}>
          {!isvalid ? (
            <AlertError variant={"danger"} message={errorMessage}></AlertError>
          ) : (
            ""
          )}
        </Col>
      </Row>
    </Form.Group>
  );
}

InputTextField.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  changeHandler: PropTypes.func,
  validation: PropTypes.object,
};
