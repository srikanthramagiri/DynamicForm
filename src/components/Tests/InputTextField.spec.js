import { shallow } from 'enzyme';
import InputTextField from '../FormComponents/InputTextField'

describe('input text field component', ()=> {
    const changeHandlerSpy = jest.fn();
    const wrapper = shallow(<InputTextField changeHandler={changeHandlerSpy}/>)
    const InputText =  wrapper.find(`[data-test='input-text-field']`)
    
    it('should render correctly', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).not.toBeNull()
    })
    it('should be one element', ()=> {
        expect(InputText).not.toBeNull()
    })
    it('change handler', ()=> {
        InputText.simulate('change', {target: {value: ''}});
        const alertItem = wrapper.find('.alert-danger')
        expect(alertItem).not.toBeNull()
    })
    it('blur handler', ()=> {
        InputText.simulate('blur', {target: {value: ''}})
        expect(changeHandlerSpy).toHaveBeenCalled()
        const alertItem = wrapper.find('.alert-danger')
        expect(alertItem).not.toBeNull()
       
    })
   
})