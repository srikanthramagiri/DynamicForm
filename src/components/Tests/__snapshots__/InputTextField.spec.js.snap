// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input text field component should render correctly 1`] = `ShallowWrapper {}`;
